decode.go

```diff
@@ -39,22 +39,22 @@ type parser struct {
 
 func newParser(b []byte) *parser {
 	p := parser{}
-	if !yaml_parser_initialize(&p.parser) {
+	if !(&p.parser).initialize() {
```

Comment:
In Go, there is no need to use the pointer

This would work

```suggestion
	if !p.parser.initialize() {
```